### PR TITLE
Add practical task js-dom

### PR DESCRIPTION
**Pull request by dizerfree**
### Practical task js-dom

[Demo](https://dizerfree.github.io/practical_task_js_dom/) |
[Code base](https://github.com/dizerfree/kottans-frontend/tree/master/task_js_dom/practical_task_js_dom)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/dizerfree/js-dom/index.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+<body>
+
+<div class="container">
+    <div class="sidebar">
+        <header>
+            <h1 class="logo">Basilio</h1>
+
+            <p>окуляри для котів</p>
+
+            <a id="hamburger_menu" class="hamburger_menu">
```
Comment (AMashoshyna) [inline submissions/dizerfree/js-dom/index.html]: I anchor element does not have "href" attribute (which means, it does not link to anything), probably it shouldn't be an `<a>` i the first place. Please think of a more appropriate html element.

File: submissions/dizerfree/js-dom/js/script.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
+
+const productsData = [
+
+    {
+        id: "oval",
+        heading: "Oval",
+        text: "Зрозуміло, що це окуляри з овальними лінзами, вважаються класичними. Ця оправа давно вже стала" +
```
Comment (AMashoshyna) [inline submissions/dizerfree/js-dom/js/script.js]: Use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Multi-line_strings) for multi-line strings.

File: submissions/dizerfree/js-dom/js/script.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    allText.appendChild(cardTitle);
+
+    var cardDescription = makeElement("p", "description", product.text);
+    allText.appendChild(cardDescription);
+
+    var cardPictureP = makeElement("p", "picture");
+    listItem.appendChild(cardPictureP);
+    var cardPicture = makeElement ("img");
+    cardPicture.src = product.imgUrl;
+    cardPicture.alt = product.heading;
```
Comment (AMashoshyna) [inline submissions/dizerfree/js-dom/js/script.js]: Very nice that you have provided `alt` attribute to images!

File: submissions/dizerfree/js-dom/js/script.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    return listItem;
+};
+
+
+var cardList = document.querySelector("main");
+
+for (var i = 0; i < productsData.length; i++) {
+    var cardItem = createCard(productsData[i]);
+    cardList.appendChild(cardItem);
```
Comment (AMashoshyna) [inline submissions/dizerfree/js-dom/js/script.js]: DOM operations are expensive in terms of performance. It is better to do them in bulks, which means that adding one big element is better that adding many small elements. That's why it is advised to create container element, append several child elements and then append container to the DOM. There is a special API for that - [document.fragment](https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment)

File: submissions/dizerfree/js-dom/js/script.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+function deactivateAllProducts () {
+    products.forEach(function (product) {
+        product.classList.remove('active-class');
+    });
+}
+
+function activateMenuLink (menuLink) {
+    deactivateAllMenuLinks();
+    deactivateAllProducts();
+    var href = menuLink.getAttribute('href');
```
Comment (AMashoshyna) [inline submissions/dizerfree/js-dom/js/script.js]: It is better to use data-* attributes to store data. One unfortunate side-effect of using href attribute for example is that anytime the user clicks the link, this value is added to browser history and when the user clicks back button, the browser attempts to retrieve this state, but nothing happens, which is confusing. Bottom line: use attributes as they were intended.

File: submissions/dizerfree/js-dom/index.html [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    </div>
+
+    <main id="product">
+
+    </main>
+</div>
+<script src="js/script.js"></script>
+<script>
+    /*
+     * Open the drawer when the menu icon is clicked.
```
Comment (AMashoshyna) [inline submissions/dizerfree/js-dom/index.html]: Why did you decide to put this script in index.html and not script.js?

File: submissions/dizerfree/js-dom/index.html [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        drawer.classList.toggle('open');
+        e.stopPropagation();
+    });
+    main.addEventListener('click', function() {
+        drawer.classList.remove('open');
+    });
+
+</script>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (dizerfree) [reply submissions/dizerfree/js-dom/index.html]: Thanks, I have fixed it

File: submissions/dizerfree/js-dom/index.html (hunk 6 above)
Comment (dizerfree) [reply submissions/dizerfree/js-dom/index.html]: Of course, thanks, I was inattentive

File: submissions/dizerfree/js-dom/js/script.js (hunk 5 above)
Comment (dizerfree) [reply submissions/dizerfree/js-dom/js/script.js]: Thanks for the tip, I replaced the "href" to "data-target" , <a> to <button> and "menuLink" to "menuButton"

File: submissions/dizerfree/js-dom/js/script.js (hunk 4 above)
Comment (dizerfree) [reply submissions/dizerfree/js-dom/js/script.js]: You're absolutely right. This is about code performance. I have fixed it and have changed "for" to "forEach" in addition

File: submissions/dizerfree/js-dom/js/script.js (hunk 3 above)
Comment (dizerfree) [reply submissions/dizerfree/js-dom/js/script.js]: Thanks, I remembered the accessibility

File: submissions/dizerfree/js-dom/js/script.js (hunk 2 above)
Comment (dizerfree) [reply submissions/dizerfree/js-dom/js/script.js]: Thanks, I have fixed it

File: submissions/dizerfree/js-dom/index.html (hunk 1 above)
Comment (dizerfree) [reply submissions/dizerfree/js-dom/index.html]: Thank you very much! I replaced the links to the buttons without the "href"

File: submissions/dizerfree/js-dom/index.html [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    <link rel="stylesheet" type="text/css" href="css/style.css">
+    <link rel="stylesheet" type="text/css" href="css/responsive.css">
+</head>
+<body>
+
+<div class="container">
+    <div class="sidebar">
+        <header>
+            <h1 class="logo">Basilio</h1>
+
```
[the remark this answers is not in the document]
Comment (dizerfree) [reply submissions/dizerfree/js-dom/index.html]: Thanks, I have fixed it

**Comment by dizerfree** [comment]
I have made all the changes. Now I get lost how to update them.
There are problems with branches again(((

**Comment by AMashoshyna** [comment]
Oh, yes. Looks weird. Can you show what `git branch -a` and `git status` gives you?
You can DM me in Telegram too.

**Comment by dizerfree** [comment]
Thank you for your patience, I figured it out

I have had problems with branches and repositories.
It turns out that the ```kottans-frontend repository``` with my impressions about courses contained also a fork of ```frontend-2019-homeworks```
In the end, I separated them and then everything worked

**Comment by AMashoshyna** [comment]
Well done!

**Comment by dizerfree** [comment]
Thanks a lot, I'm really happy. 
Tell me please, what should I do next with this PR? Close it and remove the branch?

**Comment by AMashoshyna** [comment]
The PR is merged into master, so you don't need to close it (it is no longer open). You can remove your local branch and pull the changes from master.

**Pull request merged**